3702/0/about_control_statements.rb

```diff
@@ -0,0 +1,135 @@
+# rubocop:disable Metrics/ClassLength, Lint/LiteralAsCondition, Metrics/MethodLength
+
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+class AboutControlStatements < Neo::Koan
+  def test_if_then_else_statements
+    result = if true
+               :true_value
+             else
+               :false_value
+             end
+    assert_equal :true_value, result
+  end
+
+  def test_if_then_statements
+    result = :default_value
+    result = :true_value if true
+    assert_equal :true_value, result
+  end
+
+  def test_if_statements_return_values
+    value = if true
+              :true_value
+            else
+              :false_value
+            end
+    assert_equal :true_value, value
+
+    value = if false
+              :true_value
+            else
+              :false_value
+            end
+    assert_equal :false_value, value
+  end
+
+  def test_if_statements_with_no_else_with_false_condition_return_value
+    value = (:true_value if false)
+    assert_equal nil, value
+  end
+
+  def test_condition_operators
+    assert_equal :true_value, (true ? :true_value : :false_value)
+    assert_equal :false_value, (false ? :true_value : :false_value)
+  end
+
+  def test_if_statement_modifiers
+    result = :default_value
+    result = :true_value if true
+
+    assert_equal :true_value, result
+  end
+
+  def test_unless_statement
+    result = :default_value
+    result = :false_value unless false # same as saying 'if !false', which evaluates as 'if true'
+    assert_equal :false_value, result
+  end
+
+  def test_unless_statement_evaluate_true
+    result = :default_value
+    result = :true_value unless true # same as saying 'if !true', which evaluates as 'if false'
+    assert_equal :default_value, result
+  end
+
+  def test_unless_statement_modifier
+    result = :default_value
+    result = :false_value unless false
+
+    assert_equal :false_value, result
+  end
+
+  def test_while_statement
+    i = 1
+    result = 1
+    while i <= 10
+      result *= i
+      i += 1
+    end
+    assert_equal 3_628_800, result
+  end
+
+  def test_break_statement
+    i = 1
+    result = 1
+    loop do
+      break unless i <= 10
```

Comment:
It's easier to read `break if` instead of `break unless`, just because any negation is an additional step of perception